### PR TITLE
Integrate starky in dendreth

### PR DESCRIPTION
**Pull request by Dimo99**

### CODE REVIEW

**Comment by smanilov** [comment]
I ran most binaries and they seem to run successfully, although it is unclear why some run faster and others don't.

I took a look at the code and it seems this PR mainly pipes things together and adds some serialization/deserialization code. There is a lot of repetition, which made it possible to go through the large volume of code in a rather short time.

Given that we are short on time, I'd say this PR could be merged as it is. My biggest wish would be to have the binaries as tests instead, so that testing the PR could be automated.

Please wait for @NikolayKostadinov21 's review before merging, as he can give more in-depth comments.

**Comment by NikolayKostadinov21** [comment]
I also believe that merging this PR is the best decision since I don't spot any issues related to the logic or functional inaccuracies. In a follow-up branch, I will address all the improvements that I believe should be applied.

**Review by NikolayKostadinov21 — DISMISSED** [review]

**Review by kkirkov — DISMISSED** [review]

**Review by kkirkov — APPROVED** [review]

**Review by EmilIvanichkovv — APPROVED** [review]

**Review by NikolayKostadinov21 — APPROVED** [review]

**Pull request merged**
